docs/site/submitting_a_pr.md

```diff
@@ -0,0 +1,429 @@
+properly formatted message.
+
+### 5. Push commits to the remote feature branch
+
+It is now time to push your local committed changes of your local feature branch
+directory to your remote feature branch; to keep them in synch.
+
+The very `first` time you push your changes you should run:
+
+```
+git push --set-upstream origin emonddr-doc-changes
+```
+
+The `--set-upstream` parameter will set up the tracking between your local
+feature branch and your remote feature branch.
+
+Afterwards, for any additional push, you just need to run:
+
+```
+git push
+```
+
+It is not necessary to create a pull request immediately on the push of your
+first commit; this can be done later.
+
+Do take some time to think about how many commits you want in the pull request
+when you eventually create it.
+
+If it makes sense to place your changes into one commit, then do so.
+
+If, however, it makes more sense to place your changes into a few logically
+separate commits, then do so.
+
+Whatever is the best way for the maintainers to understand the changes when the
+pull request is being reviewed.
+
+You can always re-organize commits or compress the number of commits later
+through a step called `Squashing Commits`.
+
+### 6. Rebasing
+
+Eventually your fork of the original repository will become stale, and it will
+be necessary to bring it up-to-date before actually creating a pull request.
+
+In my case, the fork of the repository is behind by several commits.
+
+![submit_pr_rebase_1.png](./imgs/submit_pr_rebase_1.png)
+
+It is necessary to perform a `rebase` .
+
+To `rebase` your forked repository's `master` branch off of the original
+repository, run :
+
+```
+git remote add upstream git@github.com:strongloop/loopback-next.git
+git checkout master
+git pull --rebase upstream master
+git push --force-with-lease origin master
+```
+
+To `rebase` your `feature` branch off of your `master` branch, run :
+
+```
+git checkout { your feature branch }
+git rebase master
+git push --force-with-lease origin { your feature branch }
+```
+
+Now we can see that the copy of the repository is no longer behind in several
+commits.
+
+The `master` branch the forked repository is `even` with original repository.
+
+![submit_pr_rebase_3.png](./imgs/submit_pr_rebase_3.png)
+
+The `feature` branch of the forked repository is 1 commit ahead of the `master`
+branch, and not `behind` on any number of commits.
+
+![submit_pr_rebase_2.png](./imgs/submit_pr_rebase_2.png)
+
+### 7. Squashing commits
+
+Squashing commits is the process of compressing many commits into one or a
+limited few, and is a great way to keep repositories concise and clean.
+
+It is a **recommended** step `before` actually creating a pull request.
+
+It a **necessary** step `after` the pull request has been created,
+reviewed, approved and is about to be merged by the maintainers.
+
+Before creating a pull request, you can choose to squash your commits into one
```

Comment:
similar for this paragraph and next... I'm thinking whether we can just skip this and just show the how.